crates/red_knot_python_semantic/src/types/call.rs

```diff
@@ -239,24 +182,12 @@ impl<'db> CallOutcome<'db> {
             } => {
                 let mut not_callable = vec![];
                 let mut union_builder = UnionBuilder::new(context.db());
-                let mut revealed = false;
```

Comment:
I don't think this is even possible or at least relevant. The only example for this that I can come up with is

```py
if coinflip():
	a = reveal_type
else:
	a = something_else

a(a) 
```

We, obviously could still support this by moving the union case handling into `infer_call_expression` too, but I honestly don't think it's worth the complexity (and there's no test)